whisper/whisperv6/filter.go

```diff
@@ -35,23 +35,29 @@ type Filter struct {
 	PoW        float64           // Proof of work as described in the Whisper spec
 	AllowP2P   bool              // Indicates whether this filter is interested in direct peer-to-peer messages
 	SymKeyHash common.Hash       // The Keccak256Hash of the symmetric key, needed for optimization
+	id         string            // unique identifier
 
 	Messages map[common.Hash]*ReceivedMessage
 	mutex    sync.RWMutex
 }
 
 // Filters represents a collection of filters
 type Filters struct {
-	watchers map[string]*Filter
-	whisper  *Whisper
-	mutex    sync.RWMutex
+	watchers         map[string]*Filter
+	topicMatcher     map[TopicType]map[*Filter]struct{}
+	allTopicsMatcher map[*Filter]struct{}
```

Comment:
// List all the filters that will be notified of a new message, no matter what its topic is.